From d00f84ad858cc457d0e894218931ef6b28cc6d4d Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Mon, 17 May 2021 13:47:50 +0200
Subject: [PATCH] Restore PaymentAddress and AddressErrors interface

Still used in Contact API:
https://github.com/WICG/contact-api/pull/39

Still referenced by Basic Card Payment Method:
https://github.com/w3c/payment-method-basic-card/issues/89
---
 ed/idl/payment-request.idl | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/ed/idl/payment-request.idl b/ed/idl/payment-request.idl
index b66bb1592..cd16e2aab 100644
--- a/ed/idl/payment-request.idl
+++ b/ed/idl/payment-request.idl
@@ -103,3 +103,31 @@ interface PaymentRequestUpdateEvent : Event {
 };
 
 dictionary PaymentRequestUpdateEventInit : EventInit {};
+
+[SecureContext, Exposed=(Window)]
+interface PaymentAddress {
+  [Default] object toJSON();
+  readonly attribute DOMString city;
+  readonly attribute DOMString country;
+  readonly attribute DOMString dependentLocality;
+  readonly attribute DOMString organization;
+  readonly attribute DOMString phone;
+  readonly attribute DOMString postalCode;
+  readonly attribute DOMString recipient;
+  readonly attribute DOMString region;
+  readonly attribute DOMString sortingCode;
+  readonly attribute FrozenArray<DOMString> addressLine;
+};
+
+dictionary AddressErrors {
+  DOMString addressLine;
+  DOMString city;
+  DOMString country;
+  DOMString dependentLocality;
+  DOMString organization;
+  DOMString phone;
+  DOMString postalCode;
+  DOMString recipient;
+  DOMString region;
+  DOMString sortingCode;
+};
-- 
2.31.1.windows.1

